engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/plan/NadelExecutionPlanFactory.kt

```diff
@@ -74,6 +75,7 @@ internal class NadelExecutionPlanFactory(
             return NadelExecutionPlanFactory(
                 executionBlueprint,
                 transforms = listOfTransforms(
+                    NadelServiceTypeFilterTransform(),
```

Comment:
does the order of transforms matter?

Reply:
Yes a bit. I figured it would be best to put it at the top so that transforms don't actually get the other types that won't end up getting executed. As for whether it's the correct decision, who knows.

I think we may need another layer of query transformation that effectively fixes up issues inside other transforms. Lest we push all that responsibility/rules onto our transform impls.